caffe2/python/rnn_cell.py

```diff
@@ -288,6 +289,31 @@ def _apply(
             raise RuntimeError(
                 'BasicRNNCell with unknown activation function (%s)'
                 % self.activation)
+        if seq_lengths is not None:
+            # TODO If this codepath becomes popular, it may be worth
+            # taking a look at optimizing it - for now a simple
+            # implementation is used to round out compatibility with
+            # ONNX.
+            if scope.CurrentDeviceScope().device_type == caffe2_pb2.CUDA:
```

Comment:
hm, it's quite hacky (and sometimes people move it to GPU after the fact). The problem is that it's literally 1 scalar but you'd need to do a memcopy for it :-\

But in this case you can utilize `CopyFromCPUInput` - it's a magic op which would do the right copy depending on the device you're running on

Reply:
switched to CopyFromCPUInput